File: baekjoon/StringQuestion/035-5635-생일.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Comparator;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        ArrayList<Integer> birthList = new ArrayList<>();
        HashMap<Integer, String> map = new HashMap<>();

        int N = Integer.parseInt(br.readLine());
        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String name = st.nextToken();
            String day = String.format("%02d", Integer.parseInt(st.nextToken()));
            String month = String.format("%02d", Integer.parseInt(st.nextToken()));
            String year = st.nextToken();
            int birth = Integer.parseInt(year + month + day);
            birthList.add(birth);
            map.put(birth, name);
        }
        
        birthList.sort(Comparator.naturalOrder());
        System.out.println(map.get(birthList.get(birthList.size()-1)));
        System.out.print(map.get(birthList.get(0)));
    }
}

File: baekjoon/StringQuestion/035-5635-생일.py
import sys


input = sys.stdin.readline
N = int(input())

map = dict()
birthList = list()
for _ in range(N):
    name, day, month, year = input().split()
    day = day.zfill(2)
    print("day:", day)
    month = month.zfill(2)
    print("month:", month)
    birthday = int(year + month + day)
    birthList.append(birthday)
    map[birthday] = name

birthList.sort()
print(map[birthList[len(birthList)-1]])
print(map[birthList[0]])